README.md

```diff
@@ -1,54 +1,202 @@
+*p_int = new_value;
+```
+However, if you want to change where `p_int` points to, you don't dereference
+it, but just treat it as a regular C++ variable:
+```cpp
+p_int = new_pointer_value;
+```
+
+The confusing thing about the ampersand in C++ is that like the asterisk, it
+also has two jobs. The one that we've already seen is to name a reference type.
+For example, a reference to an `int` would be written as `int&`. The other job
+of an ampersand is to create a pointer. So, one way to get a pointer in C++ is
+by using the `&` operator. This operator takes a variable as an operand and
+returns the address that the variable is stored in:
+```cpp
+int a = 5;
+// The next line makes b point to the location that a's value is stored at.
+// Confusingly, b also has a location where it is stored. The value at that
+// location is a's location.
+int* b = &a;
+```
+
+Another way to get pointers in C++—which is the way that you'll get `Node*`s in
+this lab—is by allocating things on the heap using the `new` keyword.
+
+The last thing that's helpful to understand is the `->` operator. As you'll
+recall, we can make classes in C++ that hold data members and have methods. The
+way you access those data members and methods is using the `.` operator. So for
+example, the `Node` class has a data member called `data` that holds an element
+of the BST. So if you had a `Node`, you could access its data by writing:
+```cpp
+node.data
+```
+The problem is that you don't have a `Node`; you have a `Node*`. This means that
+you need to dereference the pointer to get at its data. Unluckily, the `*`
+operator has lower precedence than the `.` operator. So if you were to write:
+```cpp
+*node.data
+```
+The compiler would fail to compile your code, since it's first trying to get
+`node.data`, then derefence the result of that operation. But since `Node*`s
+don't have a data member called `data`, that operation fails. You need to
+dereference the pointer first, then access the data member of the `Node` pointed
+to by the pointer:
+```cpp
+(*node).data
+```
+This is annoying to write over and over, so the creators of C++ made a
+shorthand:
+```cpp
+node->data
+```
+So all the `->` does is dereference the pointer, then access whatever is
+contained in the location pointed to by the pointer.
+
+**In summary:**
+* Pointers aren't magic. They're just regular old C++ variables. In fact,
+  they're just a number that's interpreted as an address in memory.
+* The `*`, `&`, and `->` symbols in C++ aren't magic; all three symbols
+  represent a specific operation to do with pointers, and `*` and `&` both
+  additionally enable you to name certain types in C++.
+* Both `*` and `&` can be used in C++ to name types. `T*` names a
+  pointer to a `T`, while `T&` names a reference to a `T`.
+* Both `*` and `&` can additionally be used as an operator. `*v` dereferences
+  the pointer `v`, while `&v` takes the address of `v`.
+* `v->data_member` is just a shorthand for `(*v).data_member`.
```

Comment:
Most students should know this already, so I'm not sure we need it in the main document. Can you please move this to a separate `.md` file (maybe `tips-from-the-tas.md`)? Then add a link to it in the Readme; something like
> Additional tips on C++ and the BST can be found in [Tips From the TAs](tips-from-the-tas.md)

Reply:
Yeah, this felt a bit out of place. I've just seen quite a few students putting random asterisks in their code, trying to make it compile, so this felt needed.

Reply:
Do you want the whole thing in another file, or just the part on pointers (leaving the passing pointers by reference part in the main document)?

Reply:
I'm sorry, I'm still a little confused. This bit that I wrote has two logical sections: one, recommending passing pointers by reference; and two, explaining C++ pointer syntax. The highlighted section in your review starts on line 103, which is halfway through the explanation of passing pointers by reference. In fact, the highlighted piece cuts a sentence in half. Would you like both sections moved, or only the section on C++ pointer syntax?